src/scenes/level.js

```diff
@@ -148,30 +146,55 @@ export class LevelScene extends Phaser.Scene {
 
     enemyKilled(name) {
         // Update kill quest score
-        if (characterData.getEnemiesKilled(this.currentLevel, name) < this.killQuest) {
+        if (
+            characterData.getEnemiesKilled(this.currentLevel, name) <
+            this.questAmounts[name][this.questAmounts[name].length - 1]
+        ) {
             characterData.incEnemiesKilled(this.currentLevel, name);
 
-            let questCompleted = true;
-            this.targets.forEach((enemy, index) => {
-                // Check for quest completion
-                if (
-                    characterData.getEnemiesKilled(this.currentLevel, enemy.varName) <
-                    this.killQuest
-                ) {
-                    questCompleted = false;
-                }
-                // Set as complete if all passed on last index
-                else if (questCompleted && index == this.targets.length - 1) {
-                    console.log("Quest complete!");
-                    characterData.setQuestCompleted(this.currentLevel);
-
-                    if (Math.random() < 0.5) {
-                        this.audioScene.playSfx("quest-complete-1");
-                    } else {
-                        this.audioScene.playSfx("quest-complete-2");
+            // Upgrade quest tier
+            let currentTier = characterData.getQuestTier(this.currentLevel, name);
+            console.log("enemy: %s", name);
+            console.log("tier: %s", characterData.getQuestTier(this.currentLevel, name));
+            console.log(
+                "current tier goal: %s",
+                this.questAmounts[name][currentTier - 1]
+            );
+            console.log("length: %s", this.questAmounts[name].length);
+
+            if (
+                currentTier < this.questAmounts[name].length &&
+                characterData.getEnemiesKilled(this.currentLevel, name) >=
+                    this.questAmounts[name][currentTier - 1]
+            ) {
+                characterData.incQuestTier(this.currentLevel, name);
+            }
+
+            if (!characterData.getQuestCompleted(this.currentLevel)) {
+                let questCompleted = true;
+                this.targets.forEach((enemy, index) => {
+                    // Check for level completion.
+                    // A level is considered complete when
+                    // all of the tier 1 (index 0) quests are complete.
+                    if (
+                        characterData.getEnemiesKilled(this.currentLevel, enemy.varName) <
+                        this.questAmounts[enemy.varName][0]
+                    ) {
+                        questCompleted = false;
```

Comment:
Looking at this now I'm not sure if it's necessary, quest completed should default to false

Reply:
Not sure I understand what you mean here. I think we do still need this because we need to set it to fail if any of the tier 1 quests are not completed.

Reply:
I guess it's related to the comment above on default-data. I was thinking just looing at `enemiesKilled` and calculating the tier would be simple and more readable. But you're saying it was less readable somehow?

```js
function calcTier(numKilled, tiers) {
    tiers.slice.reverse.forEach((tier, index) => {
        if (numKilled >= tier) {
            return index;
        }
    }
    return 0;
}
```

If you really think this is worse than adding/using the getters and setters and the `questTier`s then I'll believe you since you said you tried both

Reply:
Btw you can link github PRs to trello tasks and it'll show their current status which is neat. Just copy the PR link, click to expand the story and then press ctrl+v.

Reply:
Ah I like the calcTier method. That is much better.

Reply:
[Updated](https://github.com/EthanVieira/rs-clicker/pull/47/commits/fdcb36749bb7a9e1cf8f8fffdbe35ab9ec40af13)!